alica_standard_library/src/Conditions.cpp

```diff
@@ -36,4 +36,37 @@ bool AllChildSuccess(const alica::Blackboard* input, const alica::RunningPlan* r
     return true;
 }
 
+bool isFailure(const alica::RunningPlan* rp)
+{
+    if (rp->isBehaviour()) {
+        return rp->getStatus() == alica::PlanStatus::Failed;
+    } else {
+        return rp->getActiveState()->isFailureState();
+    }
+}
+
+bool AnyChildFailure(const alica::Blackboard* input, const alica::RunningPlan* rp, const alica::Blackboard* gb)
+{
+    for (const alica::RunningPlan* child : rp->getChildren()) {
+        if (isFailure(child)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+bool AllChildFailure(const alica::Blackboard* input, const alica::RunningPlan* rp, const alica::Blackboard* gb)
+{
+    for (const alica::RunningPlan* child : rp->getChildren()) {
+        if (!isFailure(child)) {
+            return false;
+        }
+    }
+    // In case of a state, make sure that all children are actually running
+    if (rp->getActiveTriple().state) {
+        return rp->getChildren().size() >= rp->getActiveTriple().state->getConfAbstractPlanWrappers().size();
```

Comment:
More for my understanding here:

is there a case where `rp->getChildren().size() < rp->getActiveTriple().state->getConfAbstractPlanWrappers().size()` so this can still return false?

Reply:
There certainly were such cases with old versions of the engine, before we simplified the threading model.  Before, every behavior ran on a separate thread, and the timing of initialization/termination calls was not deterministic.  At this point I am not so sure.